java/semantickernel-api/src/main/java/com/microsoft/semantickernel/semanticfunctions/HandlebarsPromptTemplateFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.microsoft.semantickernel.semanticfunctions;
+
+import com.microsoft.semantickernel.templateengine.handlebars.HandlebarsPromptTemplate;
+import reactor.util.annotation.NonNull;
+
+import java.util.Locale;
+
+public class HandlebarsPromptTemplateFactory implements PromptTemplateFactory {
+
+    public PromptTemplate tryCreate(@NonNull PromptTemplateConfig templateConfig) {
+        if (templateConfig.getTemplateFormat() != null &&
+                "handlebars".equals(templateConfig.getTemplateFormat().toLowerCase(Locale.ROOT))) {
```

Comment:
I would put a static constant in KernelPromptTemplateFactory of:

```
  public static final String HANDLEBARS_TEMPLATE_NAME = "handlebars";
  public static final String SEMANTIC_KERNEL_TEMPLATE_NAME = "semantic-kernel";
```

and use that, longer term it might soon make TemplateFormat enum